### PR TITLE
fix: release NGXS resources when the root module gets destroyed

### PR DESCRIPTION
**Pull request by arturovt**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

#1675 

## What is the new behavior?

This is the new behavior after the root module has been destroyed:

image

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/store/src/decorators/select/select-factory.ts [hunk 1]
```diff
@@ -9,12 +9,16 @@ import { NgxsConfig } from '../../symbols';
...
   public static store: Store | null = null;
   public static config: NgxsConfig | null = null;
 
   constructor(store: Store, config: NgxsConfig) {
     SelectFactory.store = store;
     SelectFactory.config = config;
   }
+
+  ngOnDestroy(): void {
+    SelectFactory.store = SelectFactory.config = null;
```
Comment (markwhitfeld) [inline packages/store/src/decorators/select/select-factory.ts:22]: Better readability:
```suggestion
    SelectFactory.store = null;
    SelectFactory.config = null;
```

**Review by markwhitfeld — APPROVED** [review]
Fantastic work!!!

**Comment by markwhitfeld** [comment]
Can I ask you to log a related issue and reference it from this PR. 
It just makes it more searchable. Thanks!

**Comment by markwhitfeld** [comment]
Oh, and please update the changelog in this PR too ;-)

File: packages/store/src/decorators/select/select-factory.ts (hunk 1 above)
Comment (arturovt) [reply packages/store/src/decorators/select/select-factory.ts]: Fixed.

**Comment by arturovt** [comment]
@markwhitfeld updated changelog and created an appropriate issue.

**Review by markwhitfeld — APPROVED** [review]

**Review by Carniatto — APPROVED** [review]
LGTM

**Pull request merged**